python/pyspark/sql/connect/column.py

```diff
@@ -263,6 +263,22 @@ def __str__(self) -> str:
         return f"Column({self._unparsed_identifier})"
 
 
+class SQLExpression(Expression):
+    """Returns Expression which contains a string which is a SQL expression
+    and server side will parse it by Catalyst
+    """
+
+    def __init__(self, expr: str) -> None:
+        super().__init__()
```

Comment:
should we assert here that `expr` is string and not another expression?

Reply:
In this implementation, we don't need to because the caller has verified the type (thus mypy does not complain).

I think this is a good question:
Generally speaking, I think for public API, we should throw user-facing exception, for internal API, we can assert when we want to defensive check unexpected input.

So it is a question of if we want to enforce checking cross all public/private API  (by corresponding ways). I guess maybe not now but worth it at a right time (maybe before 3.4 release).

Reply:
I think that's fine. One point of having type hints is to avoid `assert`s on those types too.